Caltech-IPAC/firefly
# DM-4593: background monitor for Firefly

https://jira.lsstcorp.org/browse/DM-4593

TO TEST:
http://localhost:8080/firefly/lc.html
Noticed 'Background Monitor' button add to top-right menu bar.
- upload any Phase Folded table, click Search
- upper right corner, click on 'Download' button
- enter the number of images and description then click search
- btw.. this dialog is only for testing purposes.  No need to review.


Background Monitor:
 - support existing features except for listed TODOs.
 - send BackgroundStatus as json action.
 - convert DownloadRequest to json
 - added BackgroundCntlr and related pieces
 - resizable layout.. 1-3 columns

Also:
 - add standardized interfaces for our Flux controllers (see BackgroundCntlr.js)
 - expand CommandServices to allow for HttpCommand
 - separate VisHeader into VisHeader and VisPreview
 - restyle banner.jsx
 - convert code to use SearchServices
 - add LightCurveFileGroupsProcessor stub
 - add VO_TABLE support to DataGroup.readAnyFormat (based on file extension only, should peek into data to detect)

TODOs:
 - downloadScript dialog
 - resend email
 - retain background jobs and statuses after reload
 - email notification is not working for local env
